pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -2027,6 +2016,33 @@ func buildTracingConfig(config *meshconfig.ProxyConfig, proxyType model.NodeType
 	return tracingCfg
 }
 
+func getPilotRandomSamplingEnv() float64 {
+	f := features.TraceSampling
+	if f < 0.0 || f > 100.0 {
+		log.Warnf("PILOT_TRACE_SAMPLING out of range: %v", f)
+		return 100.0
+	}
+	return f
+}
+
+func updateTraceSamplingConfig(config *meshconfig.ProxyConfig, cfg *http_conn.HttpConnectionManager_Tracing) {
+	sampling := getPilotRandomSamplingEnv()
+
+	if config.Tracing != nil && config.Tracing.Sampling != 0.0 {
+		log.Infof("Both PILOT_TRACE_SAMPLING and MeshConfig.ProxyConfig.Tracing.Sampling set; choosing MeshConfig.ProxyConfig.Tracing.Sampling")
```

Comment:
This will log 1x for every listener for every proxy, right? We should probably not log this much. At the very least, we should stop setting PILOT_TRACE_SAMPLING by default if we are going to log